src/lib/logger/public.ts

```diff
@@ -0,0 +1,84 @@
+import setupLoggerMethod from "./setupLoggerMethod";
+import type {
+  PublicLogger,
+  PublicLoggerInternal,
+  PublicContext,
+  WithPublicLoggerArg,
+  SetPublicContextOptions,
+} from "./types";
+import { logMessage, logError, logErrorResponse } from "./logMethods";
+import { newPublicMessage } from "./logMessages";
+export type { PublicLogger, PublicContext, SetPublicContextOptions };
+
+// Regular setContext _only exists on the server_. On the client, _always_ use
+// setPublicContext. setPublicContext can also be used on the server to set context that will be
+// accessible by the client logger (so it must be serializable!). Note that "public" means "public
+// to the browser session of the current user", so things like the current user data are safe to store
+// in public context because only that user will see them.
```

Comment:
Thanks for this note! Maybe to avoid confusion we could replace `Public` with `UserSession` instead? Not sure if that's any more or less accurate but I am worried that "public" could be misinterpreted when reading it outside of this file.

Reply:
Hmm, I don't like that name because the backend logging is _also_ for a specific user session. What about replacing `Public` with `Client`?

Reply:
I like `Client`!